external/llvm-project/mlir/lib/Conversion/ArithToAMDGPU/ArithToAMDGPU.cpp

```diff
@@ -212,7 +217,12 @@ LogicalResult TruncFToFloat8RewritePattern::match(arith::TruncFOp op) const {
   if (inType && inType.getWidth() <= 8 && saturateFP8)
     // Conversion between 8-bit floats is not supported with truncation enabled.
     return failure();
-  return success(outType.isFloat8E5M2FNUZ() || outType.isFloat8E4M3FNUZ());
+
+  return success(
+      (((outType.isFloat8E5M2FNUZ() || outType.isFloat8E4M3FNUZ()) &&
+        chipset.minorVersion >= 0x40 && chipset.minorVersion < 0x50) ||
```

Comment:
Note: add an `isGfx940()` to `Chipset` that does `majorVersion == 9 && minorVersion >= 0x40 && majorVersion < 0x50`

Then perhaps add a `hasOcpFp8()` that does `(majorVersion == 9 && minorVersion >= 0x50) || majorVersion >= 12`